### PR TITLE
[rush] Relocate subspace parameter parsing to SelectionParameterSet

### PR DESCRIPTION
**Pull request by william2958**
## Summary
This MR addresses an issue where currently when the "--subspace" parameter is passed to the install or update action, i.e. `rush install --subspace my_subspace`, all projects in the entire monorepo will be selected to be installed.

## Details
This issue was resolved by moving the "--subspace" cli parameter processing from BaseInstallAction to the SelectionParameterSet, so that the install and update actions are able to receive the correct projects for the filter parameters.

### CODE REVIEW

File: libraries/rush-lib/src/cli/actions/BaseInstallAction.ts [hunk 1]
```diff
@@ -159,9 +132,9 @@ export abstract class BaseInstallAction extends BaseRushAction {
         selectedSubspaces = this.rushConfiguration.getSubspacesForProjects(
           new Set(installManagerOptions.filteredProjects)
         );
-      } else if (this._subspaceParameter.value) {
+      } else if (this._selectionParameters?.getTargetSubspace()) {
         // Selecting a single subspace
-        const selectedSubspace: Subspace = this.rushConfiguration.getSubspace(this._subspaceParameter.value);
+        const selectedSubspace: Subspace = this._selectionParameters?.getTargetSubspace();
```
Comment (iclanton) [inline libraries/rush-lib/src/cli/actions/BaseInstallAction.ts:137]: Can you refactor this so we can avoid calling the function twice? Instead, can you stick the result in a variable and check its value in the `if` statement?

File: libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts [hunk 2]
```diff
@@ -183,6 +186,13 @@ export class SelectionParameterSet {
         ' belonging to VERSION_POLICY_NAME.' +
         ' For details, refer to the website article "Selecting subsets of projects".'
     });
+
+    this._subspaceParameter = action.defineStringParameter({
+      parameterLongName: '--subspace',
+      argumentName: 'SUBSPACE_NAME',
+      description:
+        '(EXPERIMENTAL) Specifies a Rush subspace to be installed. Requires the feature to be enabled in subspaces.json.'
```
Comment (iclanton) [inline libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts:194]: Mention what the property name in the config file needs to be set.

File: libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts [hunk 3]
```diff
@@ -260,6 +279,25 @@ export class SelectionParameterSet {
...
 
+  public getTargetSubspace(): Subspace {
+    const parameterValue: string | undefined = this._subspaceParameter.value;
+    if (parameterValue && !this._rushConfiguration.subspacesFeatureEnabled) {
+      // eslint-disable-next-line no-console
+      console.log();
+      // eslint-disable-next-line no-console
+      console.log(
+        Colorize.red(
+          `The "--subspace" parameter can only be passed if "subspacesEnabled" is set to true in subspaces.json.`
```
Comment (iclanton) [inline libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts:290]:
```suggestion
          `The "$this._subspaceParameter.longName}" parameter can only be passed if "subspacesEnabled" is set to true in subspaces.json.`
```

File: libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts (hunk 2 above)
Comment (william2958) [reply libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts]: Added

File: libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts (hunk 3 above)
Comment (william2958) [reply libraries/rush-lib/src/cli/parsing/SelectionParameterSet.ts]: Changed

File: libraries/rush-lib/src/cli/test/__snapshots__/CommandLineHelp.test.ts.snap [hunk 4]
```diff
@@ -128,8 +128,9 @@ exports[`CommandLineHelp prints the help for each action: build 1`] = `
 "usage: rush build [-h] [-p COUNT] [--timeline] [-t PROJECT] [-T PROJECT]
                   [-f PROJECT] [-o PROJECT] [-i PROJECT] [-I PROJECT]
                   [--to-version-policy VERSION_POLICY_NAME]
-                  [--from-version-policy VERSION_POLICY_NAME] [-v] [-c]
-                  [--ignore-hooks] [-s] [-m]
+                  [--from-version-policy VERSION_POLICY_NAME]
+                  [--subspace SUBSPACE_NAME] [-v] [-c] [--ignore-hooks] [-s]
```
Comment (octogonz) [inline libraries/rush-lib/src/cli/test/__snapshots__/CommandLineHelp.test.ts.snap:132]: Why does `rush build` specify `a Rush subspace to be installed`?  That seems incorrect

**Comment by octogonz** [comment]
Let's close this PR.  I think it is superseded by https://github.com/microsoft/rushstack/pull/4700

**Pull request closed without merging**